src/openrct2/scripting/bindings/world/ScMap.cpp

```diff
@@ -167,6 +167,74 @@ namespace OpenRCT2::Scripting
         return result;
     }
 
+    std::vector<DukValue> OpenRCT2::Scripting::ScMap::getAllEntitiesOnTile(const std::string& type, const DukValue& tile) const
```

Comment:
Not sure if this works, but can you take a pointer to the tile instead of a generic `DukValue` object?

```suggestion
    std::vector<DukValue> OpenRCT2::Scripting::ScMap::getAllEntitiesOnTile(const std::string& type, std::shared_ptr<ScTile> tile) const
```

Reply:
I could try it, I'll have to test what happens. If it does work, would it be better to take it as a ```const ScTile&```? Since tile is never written to and the function dosen't need ownership of the tile (The C++ Core Guidelines explains what I mean better than I can: [link](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f7-for-general-use-take-t-or-t-arguments-rather-than-smart-pointers))

Reply:
> Not sure if this works, but can you take a pointer to the tile instead of a generic `DukValue` object?

Looked through ScTile's source code just now, both the member variables and the accessors for it's x/y coords are private. Having getAllEntitiesOnTile take a ScTile would require changing the visibility of ScTile's coords and/or or adding a new public method to ScTile

Reply:
Given you only require a coordinate, wouldn't the API be better as just: `getAllEntitiesOnTile(type: string, location: CoordsXY)`?

Reply:
That's true, I'll change it to that

Reply:
> Given you only require a coordinate, wouldn't the API be better as just: `getAllEntitiesOnTile(type: string, location: CoordsXY)`?

Playing with it now, I tried to see if it would take an actual CoordsXY in c++ but it dosen't, so I'll leave it as a DukValue there. In Typescript, I think technically both a CoordsXY and a Tile object would work, since both have "x" and "y" values. There could potentially be a function signature in Typescript for taking a CoordsXY and another for taking a Tile that both correspond to the same function in c++ so that users don't need to construct a whole new object, but I'm not sure if it really matters.

Reply:
You have to use `DukValue`, but you can now do `auto coords = FromDuk<CoordsXY>(dukValue)`

Reply:
Oh okay, I didn't realize there was a ```FromDuk``` function lol

Reply:
Looking at how FromDuk is implemented, I'm guessing it has defaults for missing properties instead of throwing a DukException so I could remove the outer try-catch as well